tests/cpptests/test_cpp_llapi.cpp

```diff
@@ -1155,3 +1155,31 @@ TEST_F(LLApiTest, testScore) {
   ASSERT_EQ(opt->score, 0.5);
   RediSearch_FreeIndexOptions(opt);
 }
+
+TEST_F(LLApiTest, testInfoSize) {
+  // creating the index
+  RSIndex* index = RediSearch_CreateIndex("index", NULL);
+
+  // adding field to the index
+  RediSearch_CreateNumericField(index, NUMERIC_FIELD_NAME);
+  RediSearch_CreateTextField(index, FIELD_NAME_1);
+
+  ASSERT_EQ(RediSearch_MemUsage(index), 0);
+
+  // adding document to the index
+  RSDoc* d = RediSearch_CreateDocument(DOCID1, strlen(DOCID1), 1.0, NULL);
+  RediSearch_DocumentAddFieldNumber(d, NUMERIC_FIELD_NAME, 20, RSFLDTYPE_DEFAULT);
+  RediSearch_DocumentAddFieldCString(d, FIELD_NAME_1, "TEXT", RSFLDTYPE_DEFAULT);
+  RediSearch_SpecAddDocument(index, d);
+
+  ASSERT_EQ(RediSearch_MemUsage(index), 113);
+
+  d = RediSearch_CreateDocument(DOCID2, strlen(DOCID2), 2.0, NULL);
+  RediSearch_DocumentAddFieldCString(d, FIELD_NAME_1, "TXT", RSFLDTYPE_DEFAULT);
+  RediSearch_DocumentAddFieldNumber(d, NUMERIC_FIELD_NAME, 1, RSFLDTYPE_DEFAULT);
+  RediSearch_SpecAddDocument(index, d);
+
+  ASSERT_EQ(RediSearch_MemUsage(index), 253);
+
```

Comment:
Let's add a test for deleting docs and GC
```suggestion

  // test MemUsage after deleting docs 
  int ret = RediSearch_DropDocument(index, DOCID2, strlen(DOCID2));
  ASSERT_EQ(REDISMODULE_OK, ret);
  ASSERT_EQ(RediSearch_Size(index), 125);
  RSGlobalConfig.forkGcCleanThreshold = 0;
  index->gc->callbacks.periodicCallback(RSDummyContext, index->gc->gcCtx);
  ASSERT_EQ(RediSearch_Size(index), 114);

  ret = RediSearch_DropDocument(index, DOCID1, strlen(DOCID1));
  ASSERT_EQ(REDISMODULE_OK, ret);
  ASSERT_EQ(RediSearch_Size(index), 15);
  index->gc->callbacks.periodicCallback(RSDummyContext, index->gc->gcCtx);
  ASSERT_EQ(RediSearch_Size(index), 2);
  // we have 2 left over b/c of the offset vector size which we currently do not clean
  // since the data is not maintained

```